anime_downloader/sites/animesvostfr.py

```diff
@@ -0,0 +1,247 @@
+            #fallback = self.config['fallback_servers']
+            
+            logger.debug( "_get_sources url: %s", self.url )
+            
+            #https://www1.animesvostfr.net/ajax-get-link-stream/?server=rapidvideo&filmId=45532
+            #-> https://there.to/v/-zln1apmp3g423r
+            
+            filmid = None
+            episodeid = None
+            episodenum = None
+            
+            soup = helpers.soupify(helpers.get(self.url, params={'server' : 'rapidvideo'}, cf=True))
+            
+            shortlink = soup.find( 'link', {'rel': 'shortlink'})
+            logger.debug( "shortlink: {}".format(shortlink) )
+            if shortlink:
+                #logger.debug( "shortlink2: {}".format(shortlink['href']) )
+                ids = re.findall(r'.*/?p=([^"]+)', shortlink['href'])
+                #logger.debug( "shortlink2: {}".format(ids) )
+                if ids and len(ids) > 0:
+                    filmid = ids[0]
+                
+                
+            
+
+            select = soup.select_one('div.list-episodes  select')
+            option = None
+            
+            if select:
+                option = select.find('option', {'selected': True})
+            if option:
+                episodeid = option['episodeid']
+                episodenum = option.text
+            
+            logger.debug( "_get_sources  filmid: %s episodeid: %s episodenum: %s", str(filmid), str(episodeid), str(episodenum) )
+            
+            
+            if not filmid:
+                filmid = episodeid
+            
+            episodeurl = None
+            if filmid:
+                episodeurl = helpers.get(self.link_stream_url, params={'server' : 'rapidvideo', 'filmId' : filmid}, cf=True).text.strip()
+                
+                if not episodeurl or episodeurl == '':
+                    episodeurl = helpers.get(self.link_stream_url, params={'server' : 'photoss', 'filmId' : filmid}, cf=True).text.strip()
+                    
+            else:
+                raise NotFoundError
+                
+            logger.debug( "_get_sources url:%s", episodeurl )
+            
+            
+            
+            
+            #https://youtubedownloader.cx/v/y0661ue6-8jzj34
+            #https://youtubedownloader.cx/api/source/y0661ue6-8jzj34
+            
+            
+            #https://play.comedyshow.to/embedplay/4a2faf54b99e8d1152e598b24916d0d9
+            #<iframe src="https://www.fembed.net/v/1ekm5ij-3l2rp8p"
+            #https://feurl.com/api/source/1ekm5ij-3l2rp8p
+            # POST r: https://play.comedyshow.to
+            #d: feurl.com
+            
+            episodeurl1 = episodeurl          
+            episodeurlapi1 = episodeurl1.replace('/v/', '/api/source/' )
+            
+            episodeurl2 = ''
+            episodeurlapi2 = ''
+            
+            soup = helpers.soupify(helpers.get(episodeurl, cf=True))
```

Comment:
Cfscrape is broken, so if you want to bypass Cloudflare use `sel` instead (but if it works fine, then you can remove it or leave as-is.

Reply:
OK, i didn't know what is the function of cf=True, i took model on another script, so cf=True is use Cfscrape option ? What is it ?

Reply:
cf = True for bypassing cloudflare pages (please wait 5 seconds, you might be a robot), however it's currently outdated as cloudflare updates the page often.
sel = True accomplishes the same thing, but it's never outdated as it runs in a real browser, comes at the cost of speed.

You should use neither if there's no cloudflare waiting page.